### ISSUE TITLE
Syntax warnings in BleachBit 4.6.0-1 on Ubuntu 24.03

### ISSUE THREAD

**Issue by anthonyrevolution4** [bleachbit/bleachbit#1642]
Description:

I encountered several SyntaxWarnings during the installation of BleachBit 4.6.0-1 on my Ubuntu 24.03 system. These warnings occurred in various files within the BleachBit code.

Steps to Reproduce:

Open a terminal window.
Run the following command: sudo apt install bleachbit -y
Observe the output during the installation process.
Expected Result:

The installation should complete successfully without any warnings.

Actual Result:

The installation completes successfully, but several lines containing "SyntaxWarning" appear in the output. These lines specify the file, line number, and the specific character causing the warning.

Additional Information:

Operating System: Ubuntu 24.03
BleachBit Version: 4.6.0-1
Warnings:
/usr/share/bleachbit/bleachbit/Action.py:42 (invalid escape sequence '[')
(list all other warnings with file, line number and message)This is the actual message from the Terminal 
"sudo apt install bleachbit -y
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  bleachbit
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 543 kB of archives.
After this operation, 2,958 kB of additional disk space will be used.
Get:1 http://au.archive.ubuntu.com/ubuntu noble/universe amd64 bleachbit all 4.6.0-1 [543 kB]
Fetched 543 kB in 0s (2,047 kB/s) 
Selecting previously unselected package bleachbit.
(Reading database ... 228224 files and directories currently installed.)
Preparing to unpack .../bleachbit_4.6.0-1_all.deb ...
Unpacking bleachbit (4.6.0-1) ...
Setting up bleachbit (4.6.0-1) ...
/usr/share/bleachbit/bleachbit/Action.py:42: SyntaxWarning: invalid escape sequence '\['
  return re.search('[?*\[\]]', s) is not None
/usr/share/bleachbit/bleachbit/Cleaner.py:402: SyntaxWarning: invalid escape sequence '\W'
  '$LocalAppData\\Microsoft\\Windows\WER\\ReportQueue\\*\\*',
/usr/share/bleachbit/bleachbit/Cleaner.py:419: SyntaxWarning: invalid escape sequence '\F'
  '$windir\\Microsoft.NET\Framework\*\*.log',
/usr/share/bleachbit/bleachbit/Cleaner.py:432: SyntaxWarning: invalid escape sequence '\E'
  '$windir\\system32\\LogFiles\\WMI\\RTBackup\EtwRT.*etl',
/usr/share/bleachbit/bleachbit/Memory.py:67: SyntaxWarning: invalid escape sequence '\w'
  ret = re.search('^swapoff (\w* )?(/[\w/.-]+)$', swapoff)
/usr/share/bleachbit/bleachbit/Memory.py:159: SyntaxWarning: invalid escape sequence '\s'
  if not re.search('Filename\s+Type\s+Size', line):
/usr/share/bleachbit/bleachbit/Memory.py:162: SyntaxWarning: invalid escape sequence '\s'
  ret = re.search("%s\s+\w+\s+([0-9]+)\s" % device, line)
/usr/share/bleachbit/bleachbit/Unix.py:67: SyntaxWarning: invalid escape sequence '\.'
  """Adds a filter consisting of a prefix and a postfix
/usr/share/bleachbit/bleachbit/Unix.py:652: SyntaxWarning: invalid escape sequence '\.'
  paths = re.findall('/[/a-z\.\*]+', stdout)
/usr/share/bleachbit/bleachbit/Unix.py:720: SyntaxWarning: invalid escape sequence '\d'
  cregex = re.compile("Freed space: ([\d.]+[\s]+[BkMG])")
/usr/share/bleachbit/bleachbit/Windows.py:21: SyntaxWarning: invalid escape sequence '\S'
  """
/usr/share/bleachbit/bleachbit/Windows.py:141: SyntaxWarning: invalid escape sequence '\ '
  msg = _('The file python3.dll was found in c:\ or c:\dlls, which indicates a possible attempt at DLL search-order hijackin
g.')
/usr/share/bleachbit/bleachbit/Windows.py:157: SyntaxWarning: invalid escape sequence '\g'
  for fn in glob.glob(os.path.expandvars('%TEMP%\gdbus-nonce-file-*')):
Processing triggers for gnome-menus (3.36.0-1.1ubuntu1) ...
Processing triggers for man-db (2.12.0-3) ...
Processing triggers for desktop-file-utils (0.26-1ubuntu5) ...
"

**Comment by az0** [comment]
What is your Python version?

Here's what it looks like on Ubuntu 23.10

```
$ python3 --version
Python 3.11.6
$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=23.10
DISTRIB_CODENAME=mantic
DISTRIB_DESCRIPTION="Ubuntu 23.10"
```

**Comment by anthonyrevolution4** [comment]
Updating Your Bug Report with Additional Information
Hi Andrew,

I'm writing to provide an update on the BleachBit bug I previously reported. I've updated my Python version from 3.7 to 3.11.5, and I'm still encountering issues.
System Information:

  *   Operating System: Ubuntu 24.03 Desktop
  *   Python Version: 3.11.5
Error Message:

I'm still encountering the following error messages when running BleachBit:
/usr/share/bleachbit/bleachbit/Action.py:42: SyntaxWarning: invalid escape sequence '\['
return re.search('[?*\[\]]', s) is not None
/usr/share/bleachbit/bleachbit/Cleaner.py:402: SyntaxWarning: invalid escape sequence '\W'
'$LocalAppData\\Microsoft\\Windows\WER\\ReportQueue\\*\\*',
/usr/share/bleachbit/bleachbit/Cleaner.py:419: SyntaxWarning: invalid escape sequence '\F'
'$windir\\Microsoft.NET'
Terminal Output:

I've attached a copy of my terminal output showing the update process and the error messages I'm encountering.
Explanation:
The error messages seem to indicate that there are invalid escape sequences present in the BleachBit code. These escape sequences are used to represent special characters within strings, and the syntax used here is not valid for Python 3.11.5.

Next Steps:

I would appreciate it if you could investigate this issue further and provide any suggestions on how to resolve it. I'm happy to provide any additional information that you may need.

Attached Files:

  *   Terminal-output.txt

Thank you for your time and assistance.

Sincerely,
Anthony Autore LLB Syd. Cert & Diploma Information Technology UOW

**Comment by LinuxOnTheDesktop** [comment]
I see the same on Mint 22; Mint 22 is based upon Ubuntu 24.04. That said, I think I saw the error messages not when I updated BleachBit, but when I updated (using `apt`) some bit of Python infrastructure.

**Comment by az0** [comment]
> I see the same on Mint 22; Mint 22 is based upon Ubuntu 24.04. That said, I think I saw the error messages not when I updated BleachBit, but when I updated (using `apt`) some bit of Python infrastructure.

Yes, `SyntaxWarning: invalid escape sequence` can be trigger with updating Python to 3.12. The code is fixed and awaiting release: [run from source](https://docs.bleachbit.org/dev/running-from-source-code.html) if you want it now. Otherwise, there should be a BleachBit 4.6.1 beta release  soon.

**Comment by az0** [comment]
Fix [released in 4.6.2](https://www.bleachbit.org/news/bleachbit-462)
